File: q1930/q1930.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        char_info = dict()
        for i, c in enumerate(s):
            if c not in char_info:
                char_info[c] = [i, -1]
            else:
                char_info[c][1] = i
        
        total = 0
        for char, [l, r] in char_info.items():
            unique = set()
            for i in range(l+1, r):
                unique.add(s[i])
            total += len(unique)
            
        return total

File: q1930/q1930.java
class Solution {
    public int countPalindromicSubsequence(String s) {
        Map<Character, int[]> charInfo = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (charInfo.containsKey(c)) {
                charInfo.get(c)[1] = i; 
            } else {
                charInfo.put(c, new int[]{i, -1});
            }
        }

        int total = 0; 
        for (int[] range : charInfo.values()) {
            int l = range[0];
            int r = range[1];
            Set<Character> unique = new HashSet<>(); 
            for (int i = l + 1; i < r; i++) {
                unique.add(s.charAt(i));
            }
            total += unique.size();
        }
        return total;
    }
}
